lib/queuePopulator/LogReader.js

```diff
@@ -379,6 +390,47 @@ class LogReader {
         return undefined;
     }
 
+    _processFilterEntry(batchState, record, done) {
+        const { entriesToPublish } = batchState;
+        if (!record || !record.entries) {
+            return done();
+        }
+        this._setEntryBatch(entriesToPublish);
+        return this._filterEntries(batchState, record, err => {
+            this._unsetEntryBatch();
+            if (err) {
+                this.log.error('error filtering entries from log', {
+                    method: 'LogReader._processFilterEntries',
+                    error: err,
+                });
+                return done(err);
+            }
+            return done();
+        });
+    }
+
+    _processFilterEntries(batchState, done) {
+        const { currentRecords } = batchState;
+        if (!currentRecords || !currentRecords.length > 0) {
+            return done();
+        }
+        async.eachSeries(currentRecords,
+            (rec, next) => this._processFilterEntry(batchState, rec, next),
+            err => {
+                batchState.currentRecords = [];
```

Comment:
This should not be needed I think as I don't think the batch is reused at this point before creating the next one